compiler/prelude/prelude.go

```diff
@@ -335,6 +335,10 @@ var $append = function(slice) {
 };
 
 var $appendSlice = function(slice, toAppend) {
+  if (toAppend.constructor === String) {
+    var bytes = $stringToBytes(toAppend);
+    return $internalAppend(slice, bytes, 0, bytes.length);
+  }
```

Comment:
Because this is a special case of `append`, so it fits better here. Before, `translateConversionToSlice` covered this case, but this is not used for `append` any more. Go Spec: `As a special case, append also accepts a first argument assignable to type []byte with a second argument of string type followed by .... This form appends the bytes of the string.`